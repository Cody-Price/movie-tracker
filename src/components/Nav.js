import React, { Component } from 'react';
import { NavLink } from 'react-router-dom'


const Nav = () => {
  return (
    <header>
      <NavLink to='/favorites'>Favorites</NavLink>
      <NavLink to='/'><h1>Movie Tracker</h1></NavLink>
      <NavLink to='/login'>Login/Signup</NavLink>
    </header>
  )
}

export default Nav;