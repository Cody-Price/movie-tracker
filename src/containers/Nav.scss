@import url('https://fonts.googleapis.com/css?family=Bungee+Inline');

header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  background-color: #ba0000;
  color: white;
  height: 15vh;
  font-family: Arial, Helvetica, sans-serif;
  box-shadow: #ba0000 6px 6px 30px;
  position: fixed;
  z-index: 6;
  width: 100%;
  top: 0px;
}

a {
  color: white;
  text-decoration: none;
}

button {
  cursor: pointer;
  padding: 5px 15px;
  border-radius: 10px;
  background-color: black;
  color: white;
  border: 2px solid white;
  font-weight: bold;
  transition-duration: 0.3s;
  font-size: 14px;
  outline: none;


  &:hover {
    background-color: white;
    color: black;
    border-color: black;
    transform: scale(1.2);
    transition-duration: 0.3s;
  }

  &:active {
    background-color: #00045e;
    transition-duration: 0s;
    color: white;
    border-color: white;
  }
}

.logo {
  font-size: 50px;
  transition-duration: 0.3s;
  text-shadow: 3px 3px black;
  font-family: 'Bungee Inline', cursive;

  &:hover {
    transform: scale(1.1);
    transition-duration: 0.3s;
  }

  &:active {
    color: #00045e;
    transition-duration: 0s;
  }
}

h4 {
  position: absolute;
  // left: 10px;
  // top: -13px;
  left: 25px;
  top: 40px;
}

.nav-btn {
  transform: scale(0.7);

  &:hover {
    transform: scale(0.9);
  }
}