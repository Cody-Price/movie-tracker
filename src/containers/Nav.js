import React, { Component } from 'react';
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux';
import { logout } from '../actions'
import './Nav.scss';


const Nav = (props) => {
  return (
    <header>
      <h4> 
        {props.user.name}
      </h4>
      <NavLink to='/favorites'><button>My Favorites</button></NavLink>
      <NavLink to='/'><h1 className="logo">Movie <i class="fas fa-film"></i> Tracker</h1></NavLink>
      { props.user.name ? <NavLink to='/'><button onClick={props.logout}>Sign Out</button></NavLink> : <NavLink to='/login'><button>Login/Signup</button></NavLink> }
    </header>
  )
}

const mapStateToProps = (state) => ({
  user: state.user
})

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
