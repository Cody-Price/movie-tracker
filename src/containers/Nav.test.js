import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav'
import { connect } from 'react-redux'


describe('Nav', () => {
  let wrapper
  let mockMovies

  beforeEach(() => {
    mockMovies = [{ title: 'movie1'}, {title: 'movie2'}, {title: 'movie3' }]
    wrapper = shallow(
      <Nav
        user={{name: 'Cole', id: 1}}
        logout={jest.fn()}
        fetchMovies={jest.fn()}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should welcome the user if logged in', () => {
    const h4 = wrapper.find('h4')
    expect(h4.text()).toEqual('Welcome Back Cole!')
  })

  it('should display nothing if the user is not logged in', () => {
    wrapper = shallow(
      <Nav
        user={{name: null, id: null}}
        logout={jest.fn()}
        fetchMovies={jest.fn()}
      />
    )
    const h4 = wrapper.find('h4')
    expect(h4.text()).toEqual('')
  })
})