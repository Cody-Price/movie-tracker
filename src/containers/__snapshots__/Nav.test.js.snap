// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    fetchMovies={[MockFunction]}
    logout={[MockFunction]}
    user={
      Object {
        "id": 1,
        "name": "Cole",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Welcome Back Cole!
        </h4>,
        <button
          className="nav-btn"
          onClick={[Function]}
        >
          <i
            className="fas fa-angle-left"
          />
           Previous Page
        </button>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/favorites"
        >
          <button>
            My Favorites
          </button>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          onClick={[Function]}
          to="/"
        >
          <h1
            className="logo"
          >
            Movie 
            <i
              className="fas fa-film"
            />
             Tracker
          </h1>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          <button
            onClick={[MockFunction]}
          >
            Sign Out
          </button>
        </NavLink>,
        <button
          className="nav-btn"
          onClick={[Function]}
        >
          Next Page 
          <i
            className="fas fa-angle-right"
          />
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome Back Cole!",
        },
        "ref": null,
        "rendered": "Welcome Back Cole!",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-angle-left"
            />,
            " Previous Page",
          ],
          "className": "nav-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-angle-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          " Previous Page",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button>
            My Favorites
          </button>,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Favorites",
          },
          "ref": null,
          "rendered": "My Favorites",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <h1
            className="logo"
          >
            Movie 
            <i
              className="fas fa-film"
            />
             Tracker
          </h1>,
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Movie ",
              <i
                className="fas fa-film"
              />,
              " Tracker",
            ],
            "className": "logo",
          },
          "ref": null,
          "rendered": Array [
            "Movie ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-film",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " Tracker",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            onClick={[MockFunction]}
          >
            Sign Out
          </button>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Out",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Next Page ",
            <i
              className="fas fa-angle-right"
            />,
          ],
          "className": "nav-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Next Page ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-angle-right",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Welcome Back Cole!
          </h4>,
          <button
            className="nav-btn"
            onClick={[Function]}
          >
            <i
              className="fas fa-angle-left"
            />
             Previous Page
          </button>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/favorites"
          >
            <button>
              My Favorites
            </button>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            onClick={[Function]}
            to="/"
          >
            <h1
              className="logo"
            >
              Movie 
              <i
                className="fas fa-film"
              />
               Tracker
            </h1>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <button
              onClick={[MockFunction]}
            >
              Sign Out
            </button>
          </NavLink>,
          <button
            className="nav-btn"
            onClick={[Function]}
          >
            Next Page 
            <i
              className="fas fa-angle-right"
            />
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome Back Cole!",
          },
          "ref": null,
          "rendered": "Welcome Back Cole!",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-angle-left"
              />,
              " Previous Page",
            ],
            "className": "nav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-angle-left",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " Previous Page",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button>
              My Favorites
            </button>,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Favorites",
            },
            "ref": null,
            "rendered": "My Favorites",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <h1
              className="logo"
            >
              Movie 
              <i
                className="fas fa-film"
              />
               Tracker
            </h1>,
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Movie ",
                <i
                  className="fas fa-film"
                />,
                " Tracker",
              ],
              "className": "logo",
            },
            "ref": null,
            "rendered": Array [
              "Movie ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-film",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Tracker",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              onClick={[MockFunction]}
            >
              Sign Out
            </button>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Out",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Next Page ",
              <i
                className="fas fa-angle-right"
              />,
            ],
            "className": "nav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Next Page ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-angle-right",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
